src/analysis/mean_inequalities.lean

```diff
@@ -964,6 +964,74 @@ begin
   exact lintegral_Lp_add_le_aux hpq hf hf_top hg hg_top h0 htop,
 end
 
+lemma rpow_add_rpow_le_add {p : ℝ} (a b : ennreal) (hp1 : 1 ≤ p) :
+  (a ^ p + b ^ p) ^ (1/p) ≤ a + b :=
+begin
+  let f : bool → ennreal := λ x, ite (x = tt) a 0,
+  let g : bool → ennreal := λ x, ite (x = ff) b 0,
+  let μ := @measure.count bool ⊤,
+  have h_left : (a ^ p + b ^ p) ^ (1 / p) = (∫⁻ a, (f + g) a ^ p ∂μ) ^ (1 / p),
+  { simp_rw lintegral_count,
+    have hfg_sum : (∑' x, (f + g) x ^ p) = a^p + b^p,
+    { rw [tsum_fintype, fintype.sum_bool],
+      congr;
+      { rw pi.add_apply,
+        simp only [f, g],
+        simp, }, },
+    rw hfg_sum, },
+  have h_right : a + b = (∫⁻ a, (f a)^p ∂μ) ^ (1/p) + (∫⁻ a, (g a)^p ∂μ) ^ (1/p),
+  { simp_rw lintegral_count,
+    have hf_sum : (∑' x, f x ^ p) = a^p,
+    { rw [tsum_fintype, fintype.sum_bool],
+      change ite (tt = tt) a 0 ^ p + ite (ff = tt) a 0 ^ p = a ^ p,
+      simp [ennreal.zero_rpow_of_pos (lt_of_lt_of_le zero_lt_one hp1)], },
+    have hg_sum : (∑' x, g x ^ p) = b^p,
+    { rw [tsum_fintype, fintype.sum_bool],
+      change ite (tt = ff) b 0 ^ p + ite (ff = ff) b 0 ^ p = b ^ p,
+      simp [ennreal.zero_rpow_of_pos (lt_of_lt_of_le zero_lt_one hp1)], },
+    rw [hf_sum, hg_sum, one_div, ←ennreal.rpow_mul, ←ennreal.rpow_mul,
+      _root_.mul_inv_cancel (ne_of_lt (lt_of_lt_of_le zero_lt_one hp1)).symm, ennreal.rpow_one,
+      ennreal.rpow_one], },
+  rw [h_left, h_right],
+  have hf : measurable f, from λ s hs, measurable_space.is_measurable_top,
+  have hg : measurable g, from λ s hs, measurable_space.is_measurable_top,
+  exact ennreal.lintegral_Lp_add_le hf.ae_measurable hg.ae_measurable hp1,
+end
+
+lemma add_rpow_le_rpow_add {p : ℝ} (a b : ennreal) (hp1 : 1 ≤ p) :
```

Comment:
This looks like a super-complicated proof (although it has the advantage of avoiding all the discussion that things could be `0` or top). The most direct argument I can think of is to say, for `x = a/(a+b) \le 1` and `y = b/(a+b) \le 1`, that `x^p \le x` as `x \le 1`, and similarly for `y`, so `x^p + y^p \le x + y = 1`, which gives the desired inequality after multiplying by `(a + b)^p`. Do you have a feeling if it would be longer or shorter?

Reply:
Well, the proof I used looked like the obvious one at the time... but now that you mention that other way, my proof indeed seems over-complicated.

I cannot really say whether it will take less space, because of the case distinctions and all the (simple) manipulations involved.  It will look like some by_cases followed by lines of rw about ennreal.rpow (like the 400 lines above than lemma). I'll write it.

Reply:
Done. It's longer (but indeed very much simpler).